src/main/java/backend/Logic.java

```diff
@@ -231,11 +233,26 @@ public void removeUnusedModels(Set<String> reposInUse) {
                 .forEach(models::removeRepoModelById);
     }
 
+    /**
+     * Updates events related to remaining rate.
+     * @param rateLimits The api rate limits for updating of the remaining rate.
+     * @return Returns the rateLimits instance
+     */
     public ImmutablePair<Integer, Long> updateRemainingRate(ImmutablePair<Integer, Long> rateLimits) {
         uiManager.updateRateLimits(rateLimits);
         return rateLimits;
     }
 
+    /**
+     * Updates the sync refresh rate of updating on the current data store.
+     * @param rateLimits The api rate limits for calculation of the refresh rate.
+     * @return Returns the rateLimits instance
+     */
+    public ImmutablePair<Integer, Long> updateSyncRefreshRate(ImmutablePair<Integer, Long> rateLimits) {
+        uiManager.updateSyncRefreshRate(rateLimits);
+        return rateLimits;
+    }
+
```

Comment:
Do we need two separate events? It's not clear what's the difference between the two.